tests/test_polaraveraging.py

```diff
@@ -0,0 +1,225 @@
+    assert(isinstance(pa._compute_g_x(*data), np.ndarray))
+    testing.assert_equal(3, len(pa._compute_g_x(*data)))
+    testing.assert_equal(pa._compute_g_x(*(data[:,0])), pa._compute_g_x(*data)[0])
+    
+    # test same length array input
+    testing.assert_raises(ValueError, pa._compute_g_x, data[0,0], data[1], data[2])
+    testing.assert_raises(ValueError, pa._compute_g_x, data[0], data[1,0], data[2])
+    testing.assert_raises(ValueError, pa._compute_g_x, data[0], data[1], data[2,0])
+    testing.assert_raises(ValueError, pa._compute_g_x, data[0,0], data[1,0], data[2])
+    testing.assert_raises(ValueError, pa._compute_g_x, data[0], data[1,0], data[2,0])
+    testing.assert_raises(ValueError, pa._compute_g_x, data[0,0], data[1], data[2,0])
+    
+    # test for input range
+    testing.assert_raises(ValueError, pa._compute_g_x, 0.1, 0.1, -3.15)
+    testing.assert_raises(ValueError, pa._compute_g_x, 0.1, 0.1, 2.*np.pi+0.1)
+    testing.assert_almost_equal(pa._compute_g_x(0.1, 0.1, 0.), pa._compute_g_x(0.1, 0.1, np.pi))
+
+    
+    # test for reasonable values
+    testing.assert_almost_equal(pa._compute_g_x(100., 0., 0.), 0.0)
+    testing.assert_almost_equal(pa._compute_g_x(0., 100., np.pi/4.), 0.0)
+    testing.assert_almost_equal(pa._compute_g_x(0., 0., 0.3), 0.)
+
+def test_compute_g_t():
+    data = np.array([[0.01, 0.02, 0.01], # g1
+                     [0.01, 0.02, 0.03], # g2
+                     [0.1, 1.2, 3.]]) # phi [rad]
+
+    # test that function works for scalar and vector input
+    # testing.assert(isinstance(float, pa._compute_g_t(*(data[:,0]))))
+    # testing.assert(isinstance(np.array, pa._compute_g_t(*data)))
+    testing.assert_equal(3, len(pa._compute_g_t(*data)))
+    testing.assert_equal(pa._compute_g_t(*(data[:,0])), pa._compute_g_t(*data)[0])
+    
+    # test same length array input
+    testing.assert_raises(ValueError, pa._compute_g_t, data[0,0], data[1], data[2])
+    testing.assert_raises(ValueError, pa._compute_g_t, data[0], data[1,0], data[2])
+    testing.assert_raises(ValueError, pa._compute_g_t, data[0], data[1], data[2,0])
+    testing.assert_raises(ValueError, pa._compute_g_t, data[0,0], data[1,0], data[2])
+    testing.assert_raises(ValueError, pa._compute_g_t, data[0], data[1,0], data[2,0])
+    testing.assert_raises(ValueError, pa._compute_g_t, data[0,0], data[1], data[2,0])
+    
+    # test for input range
+    testing.assert_raises(ValueError, pa._compute_g_t, 0.1, 0.1, -3.15)
+    testing.assert_raises(ValueError, pa._compute_g_t, 0.1, 0.1, 2.*np.pi+0.1)
+    testing.assert_almost_equal(pa._compute_g_t(0.1, 0.1, 0.), pa._compute_g_t(0.1, 0.1, np.pi))
+    
+    # test for reasonable values
+    testing.assert_almost_equal(pa._compute_g_t(100., 0., 0.), -100.0)
+    testing.assert_almost_equal(pa._compute_g_t(0., 100., np.pi/4.), -100.0)
+    testing.assert_almost_equal(pa._compute_g_t(0., 0., 0.3), 0.)
+
+    
+def test_compute_radial_averages():
```

Comment:
This function has no tests to ensure the function is doing what we want! We need to make sure we test this! This is the main point of unit tests!!! Odds are if we introduced a bug that caused any of these tests to fail, the code would crash anyways.

We need to test all of the allowed functionality, to 0th order, we at least need to test 
1. Pass in a list of numbers, bin edges -> compare output to the manually tabulated solution
2. Pass in a list of numbers -> compare output in default bins to manually tabulated solution
3. We should probably also test that things work when it performs unit conversions as well.